pkg/filepiece/fpiece_test.go

```diff
@@ -0,0 +1,196 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package fpiece
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"testing"
+	// "github.com/stretchr/testify/assert"
+)
+
+var readTests = []struct {
+	in     string
+	offset int64
+	len    int64
+	out    string
+}{
+	{"butts", 0, 5, "butts"},
+	{"butts", 0, 2, "bu"},
+	{"butts", 3, 2, "ts"},
+	{"butts", 0, 10, "butts"},
+	{"butts", 1, 1100, "utts"},
+}
+
+var readAtTests = []struct {
+	in     string
+	offset int64
+	out    string
+}{
+	{"butts", 0, "butts"},
+	{"butts", 1, "utts"},
+	{"butts", 2, "tts"},
+	{"butts", 100, ""},
+	{"butts", -1, ""},
+}
+
+var writeTests = []struct {
+	in     string
+	offset int64
+	len    int64
+	out    string
+}{
+	{"butts", 0, 5, "butts"},
+	{"butts", 0, 2, "bu"},
+	{"butts", 3, 2, "\x00\x00\x00bu"},
+	{"butts", 0, 10, "butts"},
+	{"butts", 1, 1100, "\x00butts"},
+}
+
+var writeAtTests = []struct {
+	in     string
+	offset int64
+	out    string
+}{
+	{"butts", 0, "butts"},
+	{"butts", 1, "\x00butt"},
+	{"butts", 3, "\x00\x00\x00bu"},
+	{"butts", 1000, ""},
+	{"butts", -11, ""},
+}
+
+func TestRead(t *testing.T) {
+
+	for _, tt := range readTests {
+		t.Run("Reads data properly", func(t *testing.T) {
+
+			tmpfilePtr, err := ioutil.TempFile("", "read_test")
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			defer os.Remove(tmpfilePtr.Name()) // clean up
+
+			if _, err := tmpfilePtr.Write([]byte(tt.in)); err != nil {
+				log.Fatal(err)
+			}
+
+			chunk, _ := NewChunk(tmpfilePtr, tt.offset, tt.len)
```

Comment:
Why not checking for error here and at all other places in the tests? In case of test failure, it would be easier to analyze the cause if all errors are checked.